scrypto/src/math/integer/bits.rs

```diff
@@ -0,0 +1,768 @@
+                    ///
+                    /// # Examples
+                    ///
+                    /// Basic usage:
+                    ///
+                    /// ```
+                    #[doc = "use scrypto::prelude::*;"]
+                    ///
+                    #[doc = "let n = " $t "::tfrom(0b01001100u8);"]
+                    ///
+                    /// assert_eq!(n.count_ones(), 3);
+                    /// ```
+                    #[inline]
+                    #[doc(alias = "popcount")]
+                    #[doc(alias = "popcnt")]
+                    #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+                    pub fn count_ones(self) -> u32 {
+                        self.0.to_vec().iter().map(|&x| x.count_ones()).sum()
+                    }
+
+                    /// Returns the number of zeros in the binary representation of `self`.
+                    ///
+                    /// # Examples
+                    ///
+                    /// Please note that this example is shared between large integer types.
+                    /// Which explains why `I8` is used here.
+                    ///
+                    /// Basic usage:
+                    ///
+                    /// ```
+                    #[doc = "use scrypto::prelude::*;"]
+                    ///
+                    #[doc = concat!("assert_eq!(", stringify!($t),
+                        "::tfrom(0i8).count_zeros(), ", stringify!(<$t>::BITS), ");")]
+                    /// ```
+                    #[inline]
+                    #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+                    pub fn count_zeros(self) -> u32 {
+                        self.0.to_vec().iter().map(|&x| x.count_zeros()).sum()
+                    }
+
+                    /// Returns the number of trailing zeros in the binary representation of `self`.
+                    ///
+                    /// # Examples
+                    ///
+                    /// Please note that this example is shared between large integer types.
+                    /// Which explains why `I8` is used here.
+                    ///
+                    /// Basic usage:
+                    ///
+                    /// ```
+                    #[doc = "use scrypto::prelude::*;"]
+                    ///
+                    #[doc = "let n = I8::tfrom(0b0101000u8);"]
+                    ///
+                    /// assert_eq!(n.trailing_zeros(), 3);
+                    /// ```
+                    #[inline]
+                    #[must_use = "this returns the result of the operation, \
+                          without modifying the original"]
+                    pub fn trailing_zeros(self) -> u32 {
+                        let mut zeros: u32 = 0;
+                        for byte in self.0.to_vec().iter() {
```

Comment:
For my understanding, the `to_vec()` always output fixed length byte array right? And is it in little-endian?